### ISSUE TITLE
[Bug] HMR doesn't work with Storybook 6.5.0

### ISSUE THREAD

**Issue by allozaur** [eirslett/storybook-builder-vite#387]
### What version of `vite` are you using?

2.9.9

### System info and storybook versions

System:
  OS: macOS 12.3.1
  CPU: Intel i7
Binaries:
  Node: 16.10.0
  npm: 7.24.0
Browsers:
  Firefox Developer Edition 101.0b8 (64-bit)
  Chrome 101.0.4951.64

### Describe the Bug

1. To even be able to use `@storybookjs/builder-vite` i had to install `@storybookjs/addon-svelte-csf`. It didn't want to work without it (and i didn't have any `.svelte` stories yet.
2. After installing `@storybookjs/addon-svelte-csf` (had to do it forcefully as there were some dependency conflicts `@storybookjs/builder-vite` stopped throwing errors.
3. Running `npm run storybook` seems to just show the latest build ran with Webpack. When changing the stories' contents, no there is no active HMR.
4. Furthermore, if i remove `.sveltekit` and `node_modules` and then run
```bash
npm i
npm run storybook
```
i get this:

![Zrzut ekranu 2022-05-19 o 10 58 41](https://user-images.githubusercontent.com/37627284/169256145-27e910df-3038-445e-9356-40d7e671478b.jpg)

Attatching repo with this issue in ZIP

[sveltekit-storybook-vite-builder.zip](https://github.com/storybookjs/builder-vite/files/8727588/sveltekit-storybook-vite-builder.zip)

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by smashercosmo** [comment]
Yeah, I'm experiencing the same behaviour. But for me it sometimes builds with vite and sometimes with webpack, but I haven't been able to find a pattern yet :)

**Comment by smashercosmo** [comment]
btw, does your preview-head.html file contain:
```html
<script>
  window.global = window
</script>
```

this might fix the 'global is not defined' error

**Comment by IanVS** [comment]
>But for me it sometimes builds with vite and sometimes with webpack

You may be seeing webpack output from the manager.  Storybook will still use webpack to build it's own UI (the sidebar, buttons, addons panel, etc), but should consistently use vite for your preview, if you've configured it to use the vite builder.

**Comment by smashercosmo** [comment]
@IanVS ok, yeah, I think you're
@allozaur seems like not working HMR is a known issue. HMR only works when you change components, not stories. It's mentioned in the docs https://github.com/storybookjs/builder-vite#known-issues

**Comment by IanVS** [comment]
That's true, @smashercosmo, but the files should still update.  They just flash instead of a smooth HMR.  I think the issue that @allozaur was having originally was that there was a cached build in `node_modules/.cache` from webpack.  And you hit the nail on the head for his other issue with global.  Another possibility is to force jest-mock 28 to be used as described here: https://github.com/storybookjs/storybook/issues/15391#issuecomment-1129008376
